lib/rb/lib/thrift/client.rb

```diff
@@ -20,17 +20,9 @@
 
 module Thrift
   module Client
-    def initialize(iprot, middlewares = [], oprot = nil)
+    def initialize(iprot, oprot = nil)
       @iprot = iprot
       @oprot = oprot || iprot
-      @middleware = case middlewares.length
-                    when 0
-                      Middleware::NOP_MIDDLEWARE
-                    when 1
-                      middlewares.first
-                    else
-                      Middleware::MultiMiddleware.new(middlewares)
-                    end
       @seqid = 0
```

Comment:
no need middleware here ?

Reply:
if you look at the modification now the implemenetation of the endpoints of a thrift service in the ruby client would be something like:

```
class Client
   def foo(v)
     @client.call_binary('foo', Foo_args.new(v), Foo_result)
   end
end
```

for a thrift servce  like:

```
servie XYZ {
   int  foo(int v)
}
```

And so the middleware can be extracted in its own client (here: https://github.com/upfluence/uthrift-ruby/blob/master/lib/uthrift/client/middleware_client.rb#L3)